src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb_client_stats.c

```diff
@@ -87,32 +80,70 @@ void grpc_grpclb_client_stats_add_call_finished(
   }
 }
 
+void grpc_grpclb_client_stats_add_call_dropped_locked(
+    char* token, grpc_grpclb_client_stats* client_stats) {
+  // Increment num_calls_started and num_calls_finished.
+  gpr_atm_full_fetch_add(&client_stats->num_calls_started, (gpr_atm)1);
+  gpr_atm_full_fetch_add(&client_stats->num_calls_finished, (gpr_atm)1);
+  // Record the drop.
+  if (client_stats->drop_token_counts == NULL) {
+    client_stats->drop_token_counts =
+        gpr_zalloc(sizeof(grpc_grpclb_dropped_call_counts));
+  }
+  grpc_grpclb_dropped_call_counts* drop_token_counts =
+      client_stats->drop_token_counts;
+  for (size_t i = 0; i < drop_token_counts->num_entries; ++i) {
+    if (strcmp(drop_token_counts->token_counts[i].token, token) == 0) {
```

Comment:
The token is sent in a protobuf string field.  The protobuf API will always return the deserialized value as a nul-terminated string, regardless of the contents of the field, so I don't see how it's possible to get a non-nul-terminated string here, either maliciously or by accident.  The only edge case I can see regarding nul bytes is that we could get a string with a nul byte in the middle, in which case our code will just ignore everything after that (and `strncmp()` would not protect against that anyway).

Conversely, although the comment you pointed to documents that the token string will not be more than 50 characters, there is nothing (at least on the client side) enforcing this.  If we only compare the first 50 characters and then the balancer suddenly starts sending longer tokens that differ only in the last character, then we would wind up generating bogus load-reports.  I think it's better to protect against that future possibility than against the impossible-as-far-as-I-can-tell possibility of a non-nul-terminated string.